packages/frontend/core/src/components/affine/auth/use-auth.ts

```diff
@@ -75,7 +77,9 @@ export const useAuth = () => {
         'email',
         {
           email: email,
-          callbackUrl: '/auth/signIn',
+          callbackUrl: subscriptionData
+            ? subscriptionData.redirectUrl
+            : '/auth/signIn',
```

Comment:
I can see this callbackURL could have some other cases that may also change its behavior based on the route params. Maybe it is better to redesign singInCloud somehow (not necessary in this PR).